boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,847 @@
+#define PWM_CHANNEL_1_DISABLE (0x00 << PWM_EN_1_SHIFT)
+
+// Configuration values used for initializing the registers
+
+// Global Configuration
+#define GCR_CONFIG                                                           \
+    (PWM_CHANNEL_0_DISABLE | PWM_CHANNEL_1_DISABLE | PARALLEL_MODE_DISABLE | \
+     TRACK_AND_HOLD_CURRENT_DISABLE | WATCHDOG_DISABLE |                     \
+     VDD_FAILURE_DETECTION_ENABLE | CSNS_FUNCTION_DISABLED |                 \
+     OVERVOLTAGE_PROTECTION_ENABLE)
+
+// Channel  Configurations
+#define RETRY_CONFIG                                               \
+    (RANDOM_CURRENT_OFFSET_SUBTRACT | AUTO_RETRY_NUMBER_INFINITE | \
+     AUTO_RETRY_PERIOD_T_AUTO_10 | AUTO_RETRY_DC_MOTOR_DISABLED)
+#define CONFR_CONFIG                                                  \
+    (SHORT_CIRCUIT_DETECTION_ENABLE | OPEN_LOAD_DETECTION_ON_ENABLE | \
+     OPEN_LOAD_DETECTION_OFF_ENABLE | DIRECT_CONTROL_ENABLE |         \
+     SWITCH_ON_DELAY_NO_DELAY | SLEW_RATE_MEDIUM)
+#define OCR_LOW_CURRENT_SENSE_CONFIG                              \
+    (PWM_EXTERNAL_CLOCK_DIVIDER_256 | PWM_CLOCK_SOURCE_EXTERNAL | \
+     THRESHOLD_ACTIVATION_TIMES_T_OCH1_T_OCM1_L |                 \
+     MEDIUM_CURRENT_THRESHOLD_SELECT_IOCM2 |                      \
+     LOW_CURRENT_THRESHOLD_SELECT_IOCL2 |                         \
+     HIGH_CURRENT_THRESHOLD_SELECT_IOCH2 | CURRENT_SENSE_RATIO_LOW)
+#define OCR_HIGH_CURRENT_SENSE_CONFIG                             \
+    (PWM_EXTERNAL_CLOCK_DIVIDER_256 | PWM_CLOCK_SOURCE_EXTERNAL | \
+     THRESHOLD_ACTIVATION_TIMES_T_OCH1_T_OCM1_L |                 \
+     MEDIUM_CURRENT_THRESHOLD_SELECT_IOCM2 |                      \
+     LOW_CURRENT_THRESHOLD_SELECT_IOCL3 |                         \
+     HIGH_CURRENT_THRESHOLD_SELECT_IOCH2 | CURRENT_SENSE_RATIO_HIGH)
+
+struct Efuse
+{
+    ExitCode (*configure_efuse)(struct Efuse *e_fuse);
+    void (*enable_channel0)(bool enable);
+    void (*enable_channel1)(bool enable);
+    ExitCode (*get_status)(enum Efuse_Status *status, struct Efuse *e_fuse);
+    ExitCode (
+        *get_channel0_faults)(enum Efuse_Fault *fault, struct Efuse *e_fuse);
+    ExitCode (
+        *get_channel1_faults)(enum Efuse_Fault *fault, struct Efuse *e_fuse);
+    bool (*is_in_fault_mode)(void);
+    bool (*is_in_failsafe_mode)(void);
+    void (*delatch_faults)(void);
+    bool (*get_channel0_current)(struct Efuse *e_fuse, float *channel0_current);
+    bool (*get_channel1_current)(struct Efuse *e_fuse, float *channel1_current);
+
+    // The current state of the watchdog-in bit (bit 15). If the watchdog is
+    // enabled its state must be alternated at least once within the watchdog
+    // timeout period.
+    bool wdin_bit_to_set;
+};
+
+// The SPI handle for the SPI device the E-Fuses are connected to
+static SPI_HandleTypeDef *efuse_spi_handle;
+
+void Io_Efuse_Init(SPI_HandleTypeDef *const hspi)
+{
+    efuse_spi_handle = hspi;
+}
+
+void Io_Efuse_Aux1Enable(bool enable)
```

Comment:
I think the function name and parameter seems a bit misleading. Should we just break this into separate functions for enabling and disabling?